Issue with @connectrpc/connect-query: Module not found: Error: Can't resolve './use-transport'

Hi there 👋 

Unfortunately after upgrading to use `@connectrpc/connect-query`, I found that my app stopped working. I am using a regular CRA setup, and it seems that despite the package exporting both CommonJS and ESM, some of the modules cannot be found, `use-transport`, `create-use-query-options` and `create-use-infinite-query-options` to name a few.

I tried using
```ts
import { useQuery } from '@connectrpc/connect-query/dist/cjs/use-query';
```
instead, but it doesn't seem to alleviate the issue.

Also it might be worth mentioning that npmjs has a [@connectrpc/connect-react-query](https://www.npmjs.com/package/@connectrpc/connect-react-query) release `1.0.0-rc.1` which might confuse some people, perhaps it might be worth removing it if [@connectrpc/connect-query](https://www.npmjs.com/package/@connectrpc/connect-query) should be used instead?

Please find the issue below:

```sh
ERROR in ./node_modules/@connectrpc/connect-query/dist/esm/use-mutation.js 16:0-47
Module not found: Error: Can't resolve './use-transport' in 'app/node_modules/@connectrpc/connect-query/dist/esm'
Did you mean 'use-transport.js'?

BREAKING CHANGE: The request './use-transport' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
ERROR in ./node_modules/@connectrpc/connect-query/dist/esm/use-query.js 15:0-67
Module not found: Error: Can't resolve './create-use-query-options' in 'app/node_modules/@connectrpc/connect-query/dist/esm'
Did you mean 'create-use-query-options.js'?

BREAKING CHANGE: The request './create-use-query-options' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
ERROR in ./node_modules/@connectrpc/connect-query/dist/esm/use-query.js 16:0-47
Module not found: Error: Can't resolve './use-transport' in 'app/node_modules/@connectrpc/connect-query/dist/esm'
Did you mean 'use-transport.js'?

BREAKING CHANGE: The request './use-transport' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
ERROR in ./node_modules/@connectrpc/connect-query/dist/esm/use-infinite-query.js 15:0-84
Module not found: Error: Can't resolve './create-use-infinite-query-options' in 'app/node_modules/@connectrpc/connect-query/dist/esm'
Did you mean 'create-use-infinite-query-options.js'?

BREAKING CHANGE: The request './create-use-infinite-query-options' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.

```

--- a/packages/connect-query/src/call-unary-method.ts
+++ b/packages/connect-query/src/call-unary-method.ts
@@ -15,7 +15,7 @@
 import type { Message, PartialMessage } from "@bufbuild/protobuf";
 import type { CallOptions, Transport } from "@connectrpc/connect";
 
-import type { MethodUnaryDescriptor } from "./method-unary-descriptor";
+import type { MethodUnaryDescriptor } from "./method-unary-descriptor.js";
 
 /**
  * Call a unary method given it's signature and input.

--- a/packages/connect-query/src/connect-query-key.ts
+++ b/packages/connect-query/src/connect-query-key.ts
@@ -14,7 +14,7 @@
 
 import type { Message, PartialMessage } from "@bufbuild/protobuf";
 
-import type { MethodUnaryDescriptor } from "./method-unary-descriptor";
+import type { MethodUnaryDescriptor } from "./method-unary-descriptor.js";
 import type { DisableQuery } from "./utils.js";
 import { disableQuery } from "./utils.js";
 

--- a/packages/connect-query/src/create-use-infinite-query-options.ts
+++ b/packages/connect-query/src/create-use-infinite-query-options.ts
@@ -22,13 +22,13 @@ import type {
   UseSuspenseInfiniteQueryOptions,
 } from "@tanstack/react-query";
 
-import { callUnaryMethod } from "./call-unary-method";
+import { callUnaryMethod } from "./call-unary-method.js";
 import {
   type ConnectQueryKey,
   createConnectQueryKey,
-} from "./connect-query-key";
-import type { MethodUnaryDescriptor } from "./method-unary-descriptor";
-import { assert, type DisableQuery, disableQuery } from "./utils";
+} from "./connect-query-key.js";
+import type { MethodUnaryDescriptor } from "./method-unary-descriptor.js";
+import { assert, type DisableQuery, disableQuery } from "./utils.js";
 
 /**
  * Options specific to connect-query

--- a/packages/connect-query/src/create-use-query-options.ts
+++ b/packages/connect-query/src/create-use-query-options.ts
@@ -20,11 +20,11 @@ import type {
   UseSuspenseQueryOptions,
 } from "@tanstack/react-query";
 
-import { callUnaryMethod } from "./call-unary-method";
-import type { ConnectQueryKey } from "./connect-query-key";
-import { createConnectQueryKey } from "./connect-query-key";
-import type { MethodUnaryDescriptor } from "./method-unary-descriptor";
-import { assert, type DisableQuery, disableQuery } from "./utils";
+import { callUnaryMethod } from "./call-unary-method.js";
+import type { ConnectQueryKey } from "./connect-query-key.js";
+import { createConnectQueryKey } from "./connect-query-key.js";
+import type { MethodUnaryDescriptor } from "./method-unary-descriptor.js";
+import { assert, type DisableQuery, disableQuery } from "./utils.js";
 
 export interface ConnectQueryOptions {
   /** The transport to be used for the fetching. */

--- a/packages/connect-query/src/use-infinite-query.ts
+++ b/packages/connect-query/src/use-infinite-query.ts
@@ -29,9 +29,9 @@ import type {
   CreateSuspenseInfiniteQueryOptions,
 } from "./create-use-infinite-query-options";
 import { createUseInfiniteQueryOptions } from "./create-use-infinite-query-options";
-import type { MethodUnaryDescriptor } from "./method-unary-descriptor";
-import { useTransport } from "./use-transport";
-import type { DisableQuery } from "./utils";
+import type { MethodUnaryDescriptor } from "./method-unary-descriptor.js";
+import { useTransport } from "./use-transport.js";
+import type { DisableQuery } from "./utils.js";
 
 /**
  * Query the method provided. Maps to useInfiniteQuery on tanstack/react-query

--- a/packages/connect-query/src/use-mutation.ts
+++ b/packages/connect-query/src/use-mutation.ts
@@ -21,9 +21,9 @@ import type {
 import { useMutation as tsUseMutation } from "@tanstack/react-query";
 import { useCallback } from "react";
 
-import type { ConnectQueryKey } from "./connect-query-key";
-import type { MethodUnaryDescriptor } from "./method-unary-descriptor";
-import { useTransport } from "./use-transport";
+import type { ConnectQueryKey } from "./connect-query-key.js";
+import type { MethodUnaryDescriptor } from "./method-unary-descriptor.js";
+import { useTransport } from "./use-transport.js";
 
 /**
  * Options for useQuery

--- a/packages/connect-query/src/use-query.ts
+++ b/packages/connect-query/src/use-query.ts
@@ -26,11 +26,11 @@ import {
 import type {
   CreateQueryOptions,
   CreateSuspenseQueryOptions,
-} from "./create-use-query-options";
-import { createUseQueryOptions } from "./create-use-query-options";
-import type { MethodUnaryDescriptor } from "./method-unary-descriptor";
-import { useTransport } from "./use-transport";
-import type { DisableQuery } from "./utils";
+} from "./create-use-query-options.js";
+import { createUseQueryOptions } from "./create-use-query-options.js";
+import type { MethodUnaryDescriptor } from "./method-unary-descriptor.js";
+import { useTransport } from "./use-transport.js";
+import type { DisableQuery } from "./utils.js";
 
 /**
  * Query the method provided. Maps to useQuery on tanstack/react-query

--- a/packages/protoc-gen-connect-query/src/generateDts.ts
+++ b/packages/protoc-gen-connect-query/src/generateDts.ts
@@ -21,7 +21,7 @@ import {
   makeJsDoc,
 } from "@bufbuild/protoplugin/ecmascript";
 
-import type { PluginInit } from "./utils";
+import type { PluginInit } from "./utils.js";
 
 const { safeIdentifier } = codegenInfo;
 

--- a/packages/protoc-gen-connect-query/src/generateTs.ts
+++ b/packages/protoc-gen-connect-query/src/generateTs.ts
@@ -21,7 +21,7 @@ import {
   makeJsDoc,
 } from "@bufbuild/protoplugin/ecmascript";
 
-import type { PluginInit } from "./utils";
+import type { PluginInit } from "./utils.js";
 
 const { safeIdentifier } = codegenInfo;
 

